homeassistant/components/camera/ring.py

```diff
@@ -0,0 +1,142 @@
+
+    for camera in ring.stickup_cams:
+        cams.append(RingCam(hass, camera, config))
+
+    async_add_devices(cams, True)
+    return True
+
+
+class RingCam(Camera):
+    """An implementation of a Ring Door Bell camera."""
+
+    def __init__(self, hass, camera, device_info):
+        """Initialize a Ring Door Bell camera."""
+        super(RingCam, self).__init__()
+        self._camera = camera
+        self._hass = hass
+        self._name = self._camera.name
+        self._ffmpeg = hass.data[DATA_FFMPEG]
+        self._ffmpeg_arguments = device_info.get(CONF_FFMPEG_ARGUMENTS)
+        self._last_video_id = None
+        self._video_url = None
+        self._expires_at = None
+        self._utcnow = None
+
+    @property
+    def name(self):
+        """Return the name of this camera."""
+        return self._name
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        attrs = {}
+
+        attrs[ATTR_ATTRIBUTION] = CONF_ATTRIBUTION
+        attrs['device_id'] = self._camera.id
+        attrs['firmware'] = self._camera.firmware
+        attrs['kind'] = self._camera.kind
+        attrs['timezone'] = self._camera.timezone
+        attrs['type'] = self._camera.family
+        attrs['video_url'] = self._video_url
+        attrs['video_id'] = self._last_video_id
+
+        return attrs
+
+    @asyncio.coroutine
+    def async_camera_image(self):
+        """Return a still image response from the camera."""
+        from haffmpeg import ImageFrame, IMAGE_JPEG
+        ffmpeg = ImageFrame(self._ffmpeg.binary, loop=self.hass.loop)
+
+        if self._video_url is None:
+            return
+
+        image = yield from asyncio.shield(ffmpeg.get_image(
+            self._video_url, output_format=IMAGE_JPEG,
+            extra_cmd=self._ffmpeg_arguments), loop=self.hass.loop)
+        return image
+
+    @asyncio.coroutine
+    def handle_async_mjpeg_stream(self, request):
+        """Generate an HTTP MJPEG stream from the camera."""
+        from haffmpeg import CameraMjpeg
+
+        if self._video_url is None:
+            return
+
+        stream = CameraMjpeg(self._ffmpeg.binary, loop=self.hass.loop)
+        yield from stream.open_camera(
+            self._video_url, extra_cmd=self._ffmpeg_arguments)
+
+        yield from async_aiohttp_proxy_stream(
+            self.hass, request, stream,
+            'multipart/x-mixed-replace;boundary=ffserver')
+        yield from stream.close()
+
+    def update(self):
+        """Update camera entity and refresh attributes."""
+        self._utcnow = dt_util.utcnow()
+
+        if not self._last_video_id or not self._video_url:
+            self._refresh_attrs()
+
+        # refresh last_video_id if a new video has been published
+        if self._last_video_id != self._camera.last_recording_id:
+            self._refresh_attrs()
+
+        if self._utcnow >= self._expires_at:
+            self._refresh_attrs()
+
+    def _refresh_attrs(self):
```

Comment:
Remove that refresh code and set `SCAN_INTERVAL` on top of the file.

Reply:
@pvizeli  I've added the `SCAN_INTERVAL` option, however, the refresh code is necessary to avoid the regenerating the URL when not necessary. 

Since upon a new motion detected, the Ring Doorbell will generate an event and a new URL must be refreshed to reflect the latest video. Another condition to renew the URL is upon its expiration. 

Do you prefer the `_refresh_attrs()` code to moved to the `update()` method instead?